mlir/lib/Gradient/Utils/GradientShape.cpp

```diff
@@ -113,17 +113,16 @@ std::vector<Type> computeBackpropTypes(func::FuncOp callee,
 
         Type diffArgType = fnType.getInput(diffArgIndices[i]);
 
-        if (auto tensorType = diffArgType.dyn_cast<RankedTensorType>()) {
+        if (auto tensorType = dyn_cast<RankedTensorType>(diffArgType)) {
             ArrayRef<int64_t> tensorShape = tensorType.getShape();
             diffArgType = tensorType.getElementType();
             backpropResTypes.push_back(RankedTensorType::get(tensorShape, diffArgType));
```

Comment:
Why are we reconstructing the same tensor type that we already have in the header? 😅

Reply:
This is a good question, I'll simplify this.